lib/src/Uncertainty/Algorithm/Analytical/Analytical.cxx

```diff
@@ -131,6 +131,19 @@ void Analytical::run()
   Point value(standardEvent.getImplementation()->getFunction().operator()(origin));
 
   result_.setIsStandardPointOriginInFailureSpace(event_.getOperator().compare(value[0], event_.getThreshold()));
+  
+  /* check if result is valid */
+  Point physicalSpaceDesignPoint(event_.getImplementation()->getAntecedent().getDistribution().getInverseIsoProbabilisticTransformation().operator()(standardSpaceDesignPoint));
+  
+  Point valuePhysicalSpaceDesignPoint(event_.getImplementation()->getFunction().operator()(physicalSpaceDesignPoint));
+  
+  const Scalar residual = std::abs(valuePhysicalSpaceDesignPoint[0] - event_.getThreshold()); 
+  
+  const Scalar limitStateTolerance = nearestPointAlgorithm_.getMaximumConstraintError();
+  
+  if (!(residual <= 1.1 * limitStateTolerance)) // 1.1 is added to prevent numerical approximation made in Cobyla for constraint satisfaction tolerance.
+  throw Exception(HERE) << "Obtained design point is not on the limit state: its image by the limit state function is " << valuePhysicalSpaceDesignPoint[0] << ", which is incompatible with the threshold: " << event_.getThreshold() << " considering the limit state tolerance of the optimization algorithm: "<< limitStateTolerance;
```

Comment:
Shouldn't this line be indented so that it is a part of the `if`?

Reply:
yes you're right!

Reply:
At this stage, launching the `run()` statement will produce an exception, whatever the residuals.